packages/suite/src/actions/suite/metadataActions.ts

```diff
@@ -672,6 +718,259 @@ export const addMetadata = (payload: MetadataAddPayload) => (dispatch: Dispatch)
+};
+type MigratedAccountFile = {
+    newFileName: AccountMetadata['fileName'];
+    newAesKey: AccountMetadata['aesKey'];
+    data: Pick<AccountMetadata, 'accountLabel' | 'addressLabels' | 'outputLabels'>;
+};
+type MigratedFile = MigratedDeviceFile | MigratedAccountFile;
+
+const migrateMetadataFiles =
+    (missingFiles: Array<MissingFileInfo>) => async (dispatch: Dispatch, getState: GetState) => {
+        // get old keys
+        await dispatch(setDeviceMetadataKey(1, true));
+        // did user confirm labeling on device? or maybe device was not connected
+        // so suite does not have keys and needs to stop here
+        if (getState().suite.device?.metadata.status !== 'enabled') {
+            throw new Error('Labeling rejected by the user');
+        }
+        // if yes, add metadata keys to accounts
+        dispatch(syncMetadataKeys());
+
+        // try fetching data for entities with missing files
+        const filesToMigrate: Array<MigratedFile> = [];
+
+        const { device } = getState().suite;
+        if (device?.metadata.status !== 'enabled') {
+            return;
+        }
+
+        const provider = dispatch(getProvider());
+        if (!provider) {
+            throw new Error('No provider found');
+        }
+
+        const missingDeviceFile = missingFiles.find(
+            ({ type }) => type === 'device',
+        ) as MissingDeviceFile;
+
+        if (missingDeviceFile) {
+            const providerResponse = await provider.getFileContent(device?.metadata.fileName);
+
+            if (providerResponse.success && providerResponse.payload) {
+                try {
+                    const data = metadataUtils.decrypt(
+                        metadataUtils.arrayBufferToBuffer(providerResponse.payload),
+                        device?.metadata.aesKey,
+                    );
+
+                    if (data?.walletLabel) {
+                        filesToMigrate.push({
+                            newAesKey: missingDeviceFile.aesKey,
+                            newFileName: missingDeviceFile.fileName,
+                            data,
+                        });
+                    }
+                } catch (err) {
+                    console.error(err);
+                }
+            }
+        }
+
+        const hasMissingAccountFiles = missingFiles.some(({ type }) => type === 'account');
+
+        if (hasMissingAccountFiles) {
+            const accounts = getState().wallet.accounts.filter(
+                a => a.deviceState === device.state && a.metadata.fileName,
+            );
+
+            const accointsWithMissingFiles = accounts.filter(({ key }) =>
+                missingFiles.find(file => {
+                    if (file.type === 'account') {
+                        return file.key === key;
+                    }
+
+                    return false;
+                }),
+            );
+
+            const getDataPromises = accointsWithMissingFiles.map(async ({ key, metadata }) => {
```

Comment:
`getDataPromises` is accessed only on 931. So I'd change it here
to 
```
await Promise.all(accointsWithMissingFiles.map...
```

Reply:
I think it's cleaner as it is :D

Reply:
ok